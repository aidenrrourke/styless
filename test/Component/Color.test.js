import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import 'jest-styled-components';

test('Creates an opaque color object from decimal red, green and blue (RGB) values.', () => {
    const Div = styled.div`
		rgb: rgb(90, 129, 32);
		rgba: rgba(90, 129, 32, 0.5);
		rgb: rgb(@r, @g, @b);
		rgba: rgba(@r, @g, @b, @a);

		rgb: rgb(rgb(90, 129, 32));
		rgba: rgba(rgba(90, 129, 32, 0.5));
		rgb: rgb(rgb(@r, @g, @b));
		rgba: rgba(rgba(@r, @g, @b, @a));
		
		rgb: rgb(rgb(500, 300, 256));
	`;
    expect(renderer.create(<Div r={90} g={129} b={32} a={0.5}/>).toJSON()).toMatchSnapshot();
});

test('Creates a hex representation of a color in #AARRGGBB', () => {
    const Div = styled.div`
		argb: argb(rgba(90, 23, 148, 0.5));
		argb: argb(rgba(@r, @g, @b, @a));
	`;
    expect(renderer.create(<Div r={90} g={23} b={148} a={0.5}/>).toJSON()).toMatchSnapshot();
});

test('Creates an opaque color object from hue, saturation and lightness (HSL) values.', () => {
    const Div = styled.div`
		hsl: hsl(90, 100%, 50%);
		hsl1: hsl(@h, @s, @l);

		@old: hsl(90, 100%, 50%);
		new: hsl(hue(@old), 45%, 90%);

		hsla: hsla(90, 100%, 50%, 0.5);
		hsla: hsla(@h, @s, @l, @a);
	`;
    expect(renderer.create(<Div h={90} s="100%" l="50%" a={0.5}/>).toJSON()).toMatchSnapshot();
});

test('Creates an opaque color object from hue, saturation and value (HSV) values.', () => {
    const Div = styled.div`
		hsv1: hsv(90, 100%, 50%);
		hsv2: hsv(@h, @s, @v);
		
		hsva1: hsva(90, 100%, 50%, 0.5);
		hsva2: hsva(@h, @s, @v, @a);
	`;
    expect(renderer.create(<Div h={90} s="100%" v="50%" a={0.5}/>).toJSON()).toMatchSnapshot();
});

test('Extracts the hue channel of a color object in the HSL color space.', () => {
    const Div = styled.div`
		hue: hue(hsl(90, 100%, 50%));
		hue: hue(#80ff00);
		hue: hue(@color);
		hue: hue(@color2);
	`;
    expect(renderer.create(<Div color="#80ff00" color2="hsl(90, 100%, 50%)"/>).toJSON()).toMatchSnapshot();
});