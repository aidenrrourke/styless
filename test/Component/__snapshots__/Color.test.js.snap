// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculates the luma (perceptual brightness) of a color object. 1`] = `
.c0 {
  luma: 44.111615679100964;
  luma: 44.111615679100964;
}

<div
  className="c0"
  color="#64c81e"
/>
`;

exports[`Calculates the value of the luma without gamma correction. 1`] = `
.c0 {
  luminance: 65.28078431372549;
  luminance: 65.28078431372549;
  luminance: 32.768393694732794;
}

<div
  className="c0"
  color="#64c81e"
/>
`;

exports[`Creates a hex representation of a color in #AARRGGBB 1`] = `
.c0 {
  argb: #805a1794;
  argb: #805a1794;
}

<div
  className="c0"
  r={90}
/>
`;

exports[`Creates an opaque color object from decimal red, green and blue (RGB) values. 1`] = `
.c0 {
  rgb: #5a8120ff;
  rgba: #5a812080;
  rgb: #5a8120ff;
  rgba: #5a812080;
  rgb: #5a8120ff;
  rgba: #5a812080;
  rgb: #5a8120ff;
  rgba: #5a812080;
  rgb: #ffffffff;
}

<div
  className="c0"
  r={90}
/>
`;

exports[`Creates an opaque color object from hue, saturation and lightness (HSL) values. 1`] = `
.c0 {
  hsl: #80ff00ff;
  hsl1: #80ff00ff;
  new: #e6f1daff;
  hsla: #80ff0080;
  hsla: #80ff0080;
}

<div
  className="c0"
/>
`;

exports[`Creates an opaque color object from hue, saturation and value (HSV) values. 1`] = `
.c0 {
  hsv1: #408000ff;
  hsv2: #408000ff;
  hsva1: #40800080;
  hsva2: #40800080;
}

<div
  className="c0"
/>
`;

exports[`Decrease the lightness of a color in the HSL color space by an absolute amount. 1`] = `
.c0 {
  darken: #4d8a0fff;
  darken: #4d8a0fff;
  darken: #4d8a0fff;
}

<div
  className="c0"
  color="hsl(90, 80%, 50%)"
/>
`;

exports[`Decrease the saturation of a color in the HSL color space by an absolute amount. 1`] = `
.c0 {
  desaturate: #80cc33ff;
  desaturate: #80cc33ff;
  desaturate: #80cc33ff;
}

<div
  className="c0"
  color="hsl(90, 80%, 50%)"
/>
`;

exports[`Decrease the transparency (or increase the opacity) of a color, making it more opaque. Has no effect on opaque colors. 1`] = `
.c0 {
  fadein: rgba(128,242,13,0.6);
  fadein: rgba(128,242,13,0.6);
  fadein: rgba(128,242,13,0.6);
  fadein: rgba(128,242,13,0.55);
  fadein: rgba(128,242,13,0.55);
  fadein: rgba(128,242,13,0.55);
  fadein: rgb(128,242,13);
  fadein: rgb(128,242,13);
}

<div
  className="c0"
  color="hsla(90, 90%, 50%, 0.5)"
  r="relative"
/>
`;

exports[`Extracts the alpha channel of a color object. 1`] = `
.c0 {
  alpha: 0.5;
  alpha: 0.5;
  alpha: 0.5;
}

<div
  className="c0"
  color="#0a141e80"
  r={10}
/>
`;

exports[`Extracts the blue channel of a color object. 1`] = `
.c0 {
  blue: 30;
  blue: 30;
  blue: 30;
}

<div
  className="c0"
  color="#0a141e"
  r={10}
/>
`;

exports[`Extracts the green channel of a color object. 1`] = `
.c0 {
  green: 20;
  green: 20;
  green: 20;
}

<div
  className="c0"
  color="#0a141e"
  r={10}
/>
`;

exports[`Extracts the hue channel of a color object in the HSL color space. 1`] = `
.c0 {
  hue: 90;
  hue: 90;
  hue: 90;
  hue: 90;
}

<div
  className="c0"
  color="#80ff00"
/>
`;

exports[`Extracts the hue channel of a color object in the HSV color space. 1`] = `
.c0 {
  hsvhue: 90;
  hsvhue: 90;
  hsvhue: 90;
  hsvhue: 90;
}

<div
  className="c0"
  color="#80ff00"
/>
`;

exports[`Extracts the lightness  channel of a color object in the HSL color space. 1`] = `
.c0 {
  lightness: 50%;
  lightness: 50%;
  lightness: 50%;
  lightness: 50%;
  lightness: 50%;
}

<div
  className="c0"
  color="#80ff00"
/>
`;

exports[`Extracts the red channel of a color object. 1`] = `
.c0 {
  red: 10;
  red: 10;
  red: 10;
}

<div
  className="c0"
  color="#0a141e"
  r={10}
/>
`;

exports[`Extracts the saturation channel of a color object in the HSL color space. 1`] = `
.c0 {
  saturation: 100%;
  saturation: 100%;
  saturation: 100%;
  saturation: 100%;
  saturation: 50%;
}

<div
  className="c0"
  color="#80ff00"
/>
`;

exports[`Extracts the saturation channel of a color object in the HSV color space. 1`] = `
.c0 {
  hsvsaturation: 100%;
  hsvsaturation: 100%;
  hsvsaturation: 100%;
  hsvsaturation: 100%;
  hsvsaturation: 50%;
}

<div
  className="c0"
  color="#80ff00"
/>
`;

exports[`Extracts the value channel of a color object in the HSV color space. 1`] = `
.c0 {
  hsvvalue: 50%;
  hsvvalue: 50%;
  hsvvalue: 50%;
  hsvvalue: 50%;
  hsvvalue: 50%;
}

<div
  className="c0"
  color="#408000"
/>
`;

exports[`Increase the lightness of a color in the HSL color space by an absolute amount. 1`] = `
.c0 {
  lighten: #b3f075ff;
  lighten: #b3f075ff;
  lighten: #b3f075ff;
}

<div
  className="c0"
  color="hsl(90, 80%, 50%)"
/>
`;

exports[`Increase the saturation of a color in the HSL color space by an absolute amount. 1`] = `
.c0 {
  saturate: #80ff00ff;
  saturate: #80ff00ff;
  saturate: #80ff00ff;
}

<div
  className="c0"
  color="hsl(90, 80%, 50%)"
/>
`;

exports[`Increase the transparency (or decrease the opacity) of a color, making it less opaque. 1`] = `
.c0 {
  fadeout: rgba(128,242,13,0.4);
  fadeout: rgba(128,242,13,0.4);
  fadeout: rgba(128,242,13,0.4);
  fadeout: rgba(128,242,13,0.45);
  fadeout: rgba(128,242,13,0.45);
  fadeout: rgba(128,242,13,0.45);
  fadeout: rgba(128,242,13,0);
  fadeout: rgba(128,242,13,0);
}

<div
  className="c0"
  color="hsla(90, 90%, 50%, 0.5)"
  r="relative"
/>
`;
